navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -520,10 +521,10 @@ private boolean validRouteResponse(Response<DirectionsResponse> response) {
    * @param destination end point
    */
   private void fetchRoute(Position origin, Position destination) {
-    NavigationRoute.Builder routeBuilder = NavigationRoute.builder()
-      .accessToken(Mapbox.getAccessToken())
-      .origin(origin)
-      .destination(destination);
+    NavigationRoute.Builder routeBuilder =
+      navigation.getNavigationRoute(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
```

Comment:
Why do we need a new `getNavigationRoute()` function instead of just using the `profile` from `NavigationRoute.Builder`?

Reply:
NavigationRoute has no reference to MapboxNavigation, this allows MapboxNavigation to set the access token along with a profile.